pkg/controller/multiarchbuildconfig/multiarchbuildconfig.go

```diff
@@ -0,0 +1,361 @@
+			client:        mgr.GetClient(),
+			architectures: architectures,
+			buildClient:   buildClient,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to construct controller: %w", err)
+	}
+
+	predicateFuncs := predicate.Funcs{
+		CreateFunc:  func(e event.CreateEvent) bool { return true },
+		DeleteFunc:  func(event.DeleteEvent) bool { return false },
+		UpdateFunc:  func(event.UpdateEvent) bool { return false },
+		GenericFunc: func(event.GenericEvent) bool { return false },
+	}
+	if err := c.Watch(source.Kind(mgr.GetCache(), &v1.MultiArchBuildConfig{}), mabcHandler(), predicateFuncs); err != nil {
+		return fmt.Errorf("failed to create watch for MultiArchBuildConfig: %w", err)
+	}
+
+	return nil
+}
+
+func mabcHandler() handler.EventHandler {
+	return handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, o ctrlruntimeclient.Object) []reconcile.Request {
+		mabc, ok := o.(*v1.MultiArchBuildConfig)
+		if !ok {
+			logrus.WithField("type", fmt.Sprintf("%T", o)).Error("Got object that was not a MultiArchBuildConfig")
+			return nil
+		}
+
+		return []reconcile.Request{
+			{NamespacedName: types.NamespacedName{Namespace: mabc.Namespace, Name: mabc.Name}},
+		}
+	})
+}
+
+type reconciler struct {
+	logger        *logrus.Entry
+	client        ctrlruntimeclient.Client
+	buildClient   BuildClient
+	architectures []string
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	logger := r.logger.WithField("request", req.String())
+	err := r.reconcile(ctx, req, logger)
+	if err != nil {
+		logger.WithError(err).Error("Reconciliation failed")
+	} else {
+		logger.Info("Finished reconciliation")
+	}
+	return reconcile.Result{}, controllerutil.SwallowIfTerminal(err)
+}
+
+func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, logger *logrus.Entry) error {
+	logger = logger.WithField("multiarchbuildconfig_name", req.Name)
+	logger.Info("Starting reconciliation")
+
+	mabc := &v1.MultiArchBuildConfig{}
+	if err := r.client.Get(ctx, req.NamespacedName, mabc); err != nil {
+		return fmt.Errorf("failed to get the MultiArchBuildConfig: %w", err)
+	}
+
+	if mabc.Status.State == v1.SuccessState || mabc.Status.State == v1.FailureState {
+		return nil
+	}
+
+	if err := r.handleMultiArchBuildConfig(ctx, mabc); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *reconciler) handleMultiArchBuildConfig(ctx context.Context, mabc *v1.MultiArchBuildConfig) error {
+	targetImageRef := fmt.Sprintf("%s/%s", mabc.Spec.BuildSpec.CommonSpec.Output.To.Namespace, mabc.Spec.BuildSpec.CommonSpec.Output.To.Name)
+	r.logger.WithField("target_image", targetImageRef).Info("Start handling builds to generate image...")
+
+	builds, err := r.createBuildsForArchitectures(ctx, mabc)
+	if err != nil {
+		return fmt.Errorf("couldn't create builds for architectures: %s: %w", strings.Join(r.architectures, ","), err)
+	}
+
+	if err := r.watchAndHandleBuildEvents(ctx, mabc, builds); err != nil {
```

Comment:
This will only give any indication of the progress when all builds are done, which I don't think will be appreciated by people watching their test results

Reply:
The error of each of the builds is being recorded in the resource conditions as well.

Reply:
But only after all of them are done, since the resource is only modified after this point.  This is why I think we should treat `Build`s the same way we treat `MultiArchBuildConfig`s.  It's similar to what the payload plugin does with `ProwJob`s.

Reply:
Yea, when I wrote payload plugin controller, I though that it would be better to use another controller to watch prowjobs. However, I was hoping that the builds in this case, should be more trivial operation, and get away with it with just a watcher. I would prefer the controller though. I will make the changes.